src/storage/txn/store.rs

```diff
@@ -112,6 +112,28 @@ impl<S: Snapshot> Store for SnapshotStore<S> {
         lower_bound: Option<Key>,
         upper_bound: Option<Key>,
     ) -> Result<StoreScanner<S>> {
+        // Check request bounds with physical bound
+        if let Some(ref l) = lower_bound {
+            if let Some(b) = self.snapshot.physical_lower_bound() {
+                if !b.is_empty() && l.as_encoded().as_slice() < b {
```

Comment:
lower_bound also can be empty.

Reply:
It is fine for `lower_bound == ""` because empty is indeed the smallest possible value (i.e. -INF) (while for `upper_bound == ""` this is not correct). So `l.as_encoded().as_slice() < ""` will never be true.